### PR TITLE
System container support

### PR DESCRIPTION
**Pull request by codificat**
Add files and documentation to support running images built from playbook2image to be run as system containers.

One of the changes (avoid `/etc/passwd` update for `root`) is kind of independent, but it is specially relevant to this use case because normal usage is expected to run under non-root.

WIP because it's probably worth waiting for a release of `atomic` that includes https://github.com/projectatomic/atomic/pull/952 and to add the corresponding label, test and update the docs. Feedback is welcome meanwhile :)

### CODE REVIEW

File: system-container/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# System Container support
+
+A system container runs using runC instead of Docker and it is managed by the
```
Comment (aweiteka) [inline system-container/README.md]: Maybe better to say the process is managed by systemd and installed using the atomic tool.

File: system-container/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+* `service.template` is the template for the systemd service.
+
+* `tmpfiles.template` is the template for systemd-tmpfiles.
+
+The base playbook2image container image already contains all the required files
+to run as a system container, so no specific action needs to be taken to build a
+system container-ready image based on playbook2image.
+
+Once a container image is built (e.g. `my/image`) you can import it into the OSTree storage:
```
Comment (aweiteka) [inline system-container/README.md]: Let's remind or clarify that this build step is where the user adds their playbooks to the base image.

File: system-container/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	   --set PLAYBOOK_FILE=myplaybook.yml \
+	   myimage
+systemctl start myimage
+```
+
+And to finally cleanup the container:
+
+    atomic uninstall myimage
+
+*NOTE:* after the next release of the `atomic` command that includes [PR 952](https://github.com/projectatomic/atomic/pull/952) we will add an `atomic.run=once` label and the `systemctl` invocation or the cleanup/uninstall won't be necessary.
```
Comment (aweiteka) [inline system-container/README.md]: PR#952 has been merged.

File: system-container/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+If the playbooks need additional files then they can use the path `/var/lib/playbook2image` in the container as it is bind mounted from the host (the source directory is controllable with `VAR_LIB_PLAYBOOK2IMAGE`).
```
Comment (aweiteka) [inline system-container/README.md]: Is there a default host path you can call out here?

**Review by aweiteka — CHANGES_REQUESTED** [review]
Great to see this capability added. Can you mention the use case in this README? Why would I choose to use a system container vs running a container manually?

File: system-container/README.md (hunk 1 above)
Comment (codificat) [reply system-container/README.md]: I believe systemd is not involved in the "run once" scenario

File: system-container/README.md (hunk 2 above)
Comment (codificat) [reply system-container/README.md]: Right, will clean up this a bit more - I took this content from contributed docs to the openshift-ansible repo and it's indeed worth customizing a bit more for playbook2image...

File: system-container/README.md (hunk 3 above)
Comment (codificat) [reply system-container/README.md]: Yes it was, but just recently (last week) there was a release of the `atomic` tool with the changes.

Now that this is actually available I will do some actual testing :) and update

**Comment by codificat** [comment]
@aweiteka I expanded the docs a bit more with your feedback.

Also added the `atomic.run="once"` label for the system container use case. I believe that for system containers coming from playbook2image we will always want to use that *run once* approach. What do you think? Is there a use case where we'd prefer systemd to manage the execution of the playbook instead? If so, should we then document how to remove that label in the `docker build`-based docs, or should we instead not add the label by default and document how to add it?

**Review by aweiteka — APPROVED** [review]

**Comment by aweiteka** [comment]
> What do you think? Is there a use case where we'd prefer systemd to manage the execution of the playbook instead?

I think it's fine as-is right now. We can get this in and get user feedback.

Is it okay with you to merge?

**Comment by codificat** [comment]
LGTM, ship it!

**Pull request merged**